climpred/utils.py

```diff
@@ -97,3 +103,52 @@ def reduce_time_series(forecast, reference, nlags):
     forecast = forecast.where(forecast.time >= imin, drop=True)
     reference = reference.where(reference.time >= imin, drop=True)
     return forecast, reference
+
+
+def assign_climpred_compute_to_attrs(
+    skill,
+    ds,
+    function_name='compute_perfect_model',
+    metadata_dict=None,
+    metric=None,
+    comparison=None,
+):
+    """Write information about prediction skill into attrs."""
+    # assign old attrs
+    skill.attrs = ds.attrs
+
+    # climpred info
+    skill.attrs[
+        'Prediction skill'
+    ]: f'calculated by climpred https://climpred.readthedocs.io/'
+    skill.attrs['skill calculated by function'] = function_name
+    if 'init' in ds.coords:
+        skill.attrs['number of initializations'] = ds.init.size
+    if 'member' in ds.coords:
+        skill.attrs['number of members'] = ds.member.size
+
+    if 'perfect_model' in function_name:
```

Comment:
this part is not really nice. I rely on the _function names. Does anyone see a nicer version to convert back to str?

Reply:
`get_comparison_function(comparison, ALL_COMPARISONS)` returns function therefore I use `function.__name__`